WordPress/openverse-frontend
# Refactor search results components and media fetching

## Fixes


Fixes #381 by @obulat
Fixes #383 by @zackkrida 

## Description


This PR extracts a `SearchGrid` component that is responsible for displaying search results metadata for images and audio: 
- results count, `SearchRating` and `SaferBrowsing` components
- `LoadMoreButton`, `MetaSearchForm` and error messages.
It also moves data fetching on `LoadMore` button click from the `SearchGridManualLoad` and `AudioResultsList` components to the `search.vue` page, which is already responsible for data fetching on query change.
This way, there are fewer components that need to interact with Vuex store.

## Testing Instructions


Note from @zackkrida: If you see a 'Your focus-trap must have at least one container with at least one tabbable node in it at all times' error, just `npm install`.

Run the app and try searching for 'all media', images and audios, changing the applied filters and tabs. There should be no errors in the console. Also, test SSR by going to the search pages directly, for example:
http://localhost:8443/search/?q=cat&license=cc0,by
http://localhost:8443/search/image?q=cat&license=cc0,by
http://localhost:8443/search/audio?q=cat&license=cc0,by

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
